mapie/tests/test_time_series_regression.py

```diff
@@ -0,0 +1,357 @@
+    _, y_pis = mapie_ts.predict(X, alpha=0.05, optimize_beta=optimize_beta)
+    y_pred_low, y_pred_up = y_pis[:, 0, 0], y_pis[:, 1, 0]
+    width_mean = (y_pred_up - y_pred_low).mean()
+
+    coverage = regression_coverage_score(y, y_pred_low, y_pred_up)
+    np.testing.assert_allclose(width_mean, WIDTHS[strategy], rtol=1e-2)
+    np.testing.assert_allclose(coverage, COVERAGES[strategy], rtol=1e-2)
+
+
+def test_results_prefit() -> None:
+    """Test prefit results on a standard train/validation/test split."""
+    X_train_val, X_test, y_train_val, y_test = train_test_split(
+        X, y, test_size=1 / 10, random_state=1
+    )
+    X_train, X_val, y_train, y_val = train_test_split(
+        X_train_val, y_train_val, test_size=1 / 9, random_state=1
+    )
+    estimator = LinearRegression().fit(X_train, y_train)
+    mapie_ts_reg = MapieTimeSeriesRegressor(estimator=estimator, cv="prefit")
+    mapie_ts_reg.fit(X_val, y_val)
+    _, y_pis = mapie_ts_reg.predict(X_test, alpha=0.05)
+    width_mean = (y_pis[:, 1, 0] - y_pis[:, 0, 0]).mean()
+    coverage = regression_coverage_score(
+        y_test, y_pis[:, 0, 0], y_pis[:, 1, 0]
+    )
+    np.testing.assert_allclose(width_mean, WIDTHS["prefit"], rtol=1e-2)
+    np.testing.assert_allclose(coverage, COVERAGES["prefit"], rtol=1e-2)
+
+
+def test_not_enough_resamplings() -> None:
+    """Test that a warning is raised if at least one residual is nan."""
+    with pytest.warns(UserWarning, match=r"WARNING: at least one point of*"):
+        mapie_ts_reg = MapieTimeSeriesRegressor(
+            cv=BlockBootstrap(n_resamplings=1, n_blocks=1), agg_function="mean"
+        )
+        mapie_ts_reg.fit(X, y)
+
+
+def test_no_agg_fx_specified_with_subsample() -> None:
+    """Test that a warning is raised if at least one residual is nan."""
+    with pytest.raises(
+        ValueError, match=r"You need to specify an aggregation*"
+    ):
+        mapie_ts_reg = MapieTimeSeriesRegressor(
+            cv=BlockBootstrap(n_resamplings=1, n_blocks=1),
+            agg_function=None,
+        )
+        mapie_ts_reg.fit(X, y)
+
+
+def test_invalid_aggregate_all() -> None:
+    """
+    Test that wrong aggregation in MAPIE raise errors.
+    """
+    with pytest.raises(
+        ValueError,
+        match=r".*Aggregation function called but not defined.*",
+    ):
+        aggregate_all(None, X)
+
+
+def test_aggregate_with_mask_with_prefit() -> None:
+    """
+    Test ``aggregate_with_mask`` in case ``cv`` is ``"prefit"``.
+    """
+    mapie_ts_reg = MapieTimeSeriesRegressor(cv="prefit")
+    with pytest.raises(
+        ValueError,
+        match=r".*There should not be aggregation of predictions if cv is*",
+    ):
+        mapie_ts_reg.aggregate_with_mask(k, k)
+
+    mapie_ts_reg = MapieTimeSeriesRegressor(agg_function="nonsense")
+    with pytest.raises(
+        ValueError,
+        match=r".*The value of self.agg_function is not correct*",
+    ):
+        mapie_ts_reg.aggregate_with_mask(k, k)
```

Comment:
Shouldn't this method hidden with a `_` prefix ?

Reply:
yes
I also delete these test in test_time_series_regression.py since they are redundant with test_regression.py